storage/innobase/mtr/mtr0log.cc

```diff
@@ -798,9 +809,36 @@ bool mlog_open_and_write_index(mtr_t *mtr, const byte *rec,
     n = DICT_INDEX_SPATIAL_NODEPTR_SIZE;
   }
 
+  /* Ordinal position of an existing field can't be changed with INSTANT
+  algorithm. But when it is combined with ADD/DROP COLUMN, ordinal position
+  of a filed can be changed. This bool array of size #fields in index,
+  represents if ordinal position of an existing filed is changed. */
+  
+  // Move the logic to find columns that changed order before calculating
+  // the buffer size
+  bool *fields_with_changed_order = nullptr;
+  if (is_versioned) {
+    fields_with_changed_order = new bool[n];
```

Comment:
`fields_with_changed_order` should be freed.
if (!mlog_open(mtr, alloc, log_ptr)) {
  if (is_versioned) {
    delete[] fields_with_changed_order;
  }
}

Reply:
Thanks for review! It's a good catch, and I've fixed this leak in latest update.